relays/ethereum/src/substrate_client.rs

```diff
@@ -341,3 +413,20 @@ fn create_unsigned_submit_transaction(header: QueuedEthereumHeader) -> bridge_no
 
 	bridge_node_runtime::UncheckedExtrinsic::new_unsigned(function)
 }
+
+/// When RPC method returns encoded value.
+fn rpc_returns_encoded_value<T: Decode>(value: Value) -> Result<T, Error> {
+	let encoded_response: Bytes = from_value(value).map_err(|_| Error::ResponseParseFailed)?;
```

Comment:
When a request fails to decode, would there be any indication about _which_ request failed?

Reply:
You'll get something like "Error retrieving best header number from Substrate: <...>" in logs. We always have at most 1 request at a time with a single exception - when we're submitting multiple headers in multiple transactions. But this would lead to single error anyway - "Error submitting headers to Substrate node: <...>".